File: app/src/main/java/ethiopia/covid/android/ui/fragment/QuestionPageFragment.java
package ethiopia.covid.android.ui.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import ethiopia.covid.android.R;
import ethiopia.covid.android.data.QuestionnaireItem;
import ethiopia.covid.android.data.QuestionItem;
import ethiopia.covid.android.ui.adapter.CheckBoxQuestionRecyclerAdapter;
import ethiopia.covid.android.ui.adapter.OnQuestionItemSelected;
import ethiopia.covid.android.ui.adapter.SingleChoiceQuestionRecyclerAdapter;

/**
 * Created by BrookMG on 3/23/2020 in ethiopia.covid.android.ui.fragment
 * inside the project CoVidEt .
 */
public class QuestionPageFragment extends BaseFragment {

    private RecyclerView mainRecyclerView;
    private TextView mainTextView;
    private ConstraintLayout mainContainer;
    private QuestionnaireItem.QuestionType type;
    private List<QuestionItem> questionItems;
    private OnQuestionItemSelected onQuestionItemSelected;
    private String questionTitle;

    public QuestionPageFragment(
            String questionTitle,
            QuestionnaireItem.QuestionType type,
            List<QuestionItem> questionItems,
            OnQuestionItemSelected onQuestionItemSelected
    ) {
        this.questionTitle = questionTitle;
        this.type = type;
        this.questionItems = questionItems;
        this.onQuestionItemSelected = onQuestionItemSelected;
    }

    public static QuestionPageFragment newInstance(
            String questionTitle,
            QuestionnaireItem.QuestionType type,
            List<QuestionItem> items,
            OnQuestionItemSelected onQuestionItemSelected
    ) {
        Bundle args = new Bundle();
        QuestionPageFragment fragment = new QuestionPageFragment(questionTitle, type, items, onQuestionItemSelected);
        fragment.setArguments(args);
        return fragment;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        View mainView = inflater.inflate(R.layout.question_page_fragment, container, false);
        mainRecyclerView = mainView.findViewById(R.id.recycler_view);
        mainTextView = mainView.findViewById(R.id.textView_1);
        mainContainer = mainView.findViewById(R.id.main_container);

        mainTextView.setText(questionTitle);

        switch (type) {
            case SINGLE_CHOICE_QUESTION:
            case SINGLE_MULTIPLE_CHOICE_QUESTION:
                mainRecyclerView.setLayoutManager(new LinearLayoutManager(getActivity() , RecyclerView.VERTICAL , false));
                mainRecyclerView.setAdapter(new CheckBoxQuestionRecyclerAdapter(questionItems, onQuestionItemSelected));
                break;
            case SINGLE_BLOCK_QUESTION:
                mainRecyclerView.setLayoutManager(new GridLayoutManager(getActivity() , 2));
                mainRecyclerView.setAdapter(new SingleChoiceQuestionRecyclerAdapter(questionItems, onQuestionItemSelected));
                break;
        }

        return mainView;
    }

}

File: app/src/main/java/ethiopia/covid/android/ui/fragment/QuestionPageFragment.kt
package ethiopia.covid.android.ui.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import ethiopia.covid.android.R
import ethiopia.covid.android.data.QuestionItem
import ethiopia.covid.android.data.QuestionnaireItem.QuestionType
import ethiopia.covid.android.databinding.QuestionPageFragmentBinding
import ethiopia.covid.android.ui.adapter.CheckBoxQuestionRecyclerAdapter
import ethiopia.covid.android.ui.adapter.OnQuestionItemSelected
import ethiopia.covid.android.ui.adapter.SingleChoiceQuestionRecyclerAdapter

/**
 * Created by BrookMG on 3/23/2020 in ethiopia.covid.android.ui.fragment
 * inside the project CoVidEt .
 */
class QuestionPageFragment(
        private val questionTitle: String?,
        private val type: QuestionType?,
        private val questionItems: List<QuestionItem>,
        private val onQuestionItemSelected: OnQuestionItemSelected?
) : BaseFragment() {

    private lateinit var questionPageFragmentBinding: QuestionPageFragmentBinding

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        questionPageFragmentBinding = QuestionPageFragmentBinding.inflate(layoutInflater)
        questionPageFragmentBinding.textView1.text = questionTitle

        when (type) {
            QuestionType.SINGLE_CHOICE_QUESTION -> {
                questionPageFragmentBinding.recyclerView.layoutManager = LinearLayoutManager(activity, RecyclerView.VERTICAL, false)
                questionPageFragmentBinding.recyclerView.adapter = CheckBoxQuestionRecyclerAdapter(questionItems, true, onQuestionItemSelected)
            }
            QuestionType.SINGLE_MULTIPLE_CHOICE_QUESTION -> {
                questionPageFragmentBinding.recyclerView.layoutManager = LinearLayoutManager(activity, RecyclerView.VERTICAL, false)
                questionPageFragmentBinding.recyclerView.adapter = CheckBoxQuestionRecyclerAdapter(questionItems, onQuestionItemSelected)
            }
            QuestionType.SINGLE_BLOCK_QUESTION -> {
                questionPageFragmentBinding.recyclerView.layoutManager = GridLayoutManager(activity, 2)
                questionPageFragmentBinding.recyclerView.adapter = SingleChoiceQuestionRecyclerAdapter(questionItems, onQuestionItemSelected)
            }
        }
        return questionPageFragmentBinding.root
    }

    companion object {
        fun newInstance(
                questionTitle: String?,
                type: QuestionType?,
                items: List<QuestionItem>,
                onQuestionItemSelected: OnQuestionItemSelected?
        ): QuestionPageFragment {
            val args = Bundle()
            val fragment = QuestionPageFragment(questionTitle, type, items, onQuestionItemSelected)
            fragment.arguments = args
            return fragment
        }
    }

}